### ISSUE TITLE
Disabling datagrid row selection not working with `*ngFor`

### ISSUE THREAD

**Issue by zhichl** [vmware-clarity/ng-clarity#157]
## Describe the bug

We (NSX-ALB team) are setting:`[clrDgSelectable]="false"` for `<clr-dg-row>` as the snippet shown:
```
    <clr-dg-row
        *ngFor="let row of rows; index as i; trackBy: trackByRowName"
        [clrDgItem]="row"
        [clrDgSelectable]="false"
    >
```
However, the leading checkbox of a row in the UI is not grayed out for selection.
<img width="540" alt="image" src="https://user-images.githubusercontent.com/5053922/172533939-0982af90-e766-4590-a7a8-b49fad955eba.png">

I found a similar bug which was given a workaround: https://github.com/vmware/clarity/issues/4993.
We’ve tried the workaround but the bug is still present. Has there been a known solution for this issue?

## How to reproduce

Please check: https://stackblitz.com/edit/clarity-light-theme-clr13-cds6-hmw4an?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts

1.  Click on the link to enter the project.
2.  Check `app.component.html` line 30 to see `[clrDgSelectable]` is set to `false`, save and look at the layout on the right side for checkboxes.
3.  Go to `app.component.html`, comment out line 22 through line 26 and comment line 27 through line 31. This will bind `[clrDgSelectable]` to a function call which always returns `false`. Save and look at the layout on the right side for checkboxes.

## Expected behavior

Under either case from the reproduction part, checkboxes are not disabled.

## Versions

**Clarity project:**

ng-clarity

**Clarity version:**
- v5.x
- v6.x

**Framework:**

Angular

**Framework version:**
_Angular 12_

**Device:**

- Type: MacBook
- OS: MacOS
- Browser: Chrome

**Comment by kevinbuhmann** [comment]
The `clrDgSelectable` input only works with the `*clrDgItems` smart iterator. I will look into into making it work it work with `*ngFor` as well.

**Comment by zhichl** [comment]
@kevinbuhmann Any updates on how this bug will be handled?

**Comment by kevinbuhmann** [comment]
Not at this time. I spent some time on this and wasn't able to figure out why it's not working in your StackBlitz. It works in our Storybook. I was going to consult with my teammate next week to see if he has any idea as he knows the datagrid better than I do.

https://storybook.clarity.design/?path=/story/datagrid-row--default&args=clrDgSelectable:false;multiSelectable:true

**Comment by Jinnie** [comment]
It's probably some timing issue. I debugged the demo a bit and I see that:
- canItBeLocked() (in the selection service) fails, because
- this._items.all is still undefined at the moment of execution
- the items service is still uninitialized
We might want to tie the setting to the Items.allChanges observable, or just bravely push the call a bit forward in time and expect the initialization to be complete.


![Screen Shot 2023-01-10 at 13 00 18](https://user-images.githubusercontent.com/1798828/211534408-1c9200e9-8b5f-4dc3-9514-b553950abb3d.jpg)

**Comment by kevinbuhmann** [comment]
I think I figured this out. The bug is related to #474 and #503.

**Comment by williamernest** [comment]
This has been resolved in #537 and #531.

**Comment by kevinbuhmann** [comment]
This is not fixed.
